File: src/main/java/me/stavgordeev/plugin/commands/DiscoMayhemCommands.java
// src/main/java/me/stavgordeev/plugin/commands/MinigameCommand.java
package me.stavgordeev.plugin.commands;

import me.stavgordeev.plugin.Constants.DiscoMayhemConst;
import me.stavgordeev.plugin.Minigames.DiscoMayhem;
import net.kyori.adventure.text.Component;
import net.kyori.adventure.text.format.NamedTextColor;
import org.bukkit.Bukkit;
import org.bukkit.command.Command;
import org.bukkit.command.CommandSender;
import org.bukkit.entity.Player;
import org.jetbrains.annotations.Nullable;

import java.util.List;

public class DiscoMayhemCommands extends MinigameCommandsSkeleton {
    private final DiscoMayhem discoMayhem;

    public DiscoMayhemCommands(DiscoMayhem discoMayhem) {
        this.discoMayhem = discoMayhem;
    }

    @Override
    protected boolean handleCommand(Player player, Command command, String label, String[] args) {
        switch (args[0].toLowerCase()) {
            case "start":
                try {
                    discoMayhem.start(player);
                } catch (InterruptedException e) {
                    throw new RuntimeException(e);
                }
                break;
            case "start_hard_mode":
                try {
                    discoMayhem.startFastMode(player);
                } catch (InterruptedException e) {
                    throw new RuntimeException(e);
                }
                break;
            case "stop":
                discoMayhem.pauseGame(player);
                break;
            case "resume":
                discoMayhem.resumeGame(player);
                break;
            case "end":
                discoMayhem.endGame(player);
                break;
            case "nuke_area":
                discoMayhem.nukeArea(DiscoMayhemConst.GAME_START_LOCATION, 50);
                break;
            default:
                Bukkit.getServer().broadcast(Component.text("Unknown command.").color(NamedTextColor.RED));
                break;
        }
        return true;
    }

    @Override
    protected @Nullable List<String> handleTabComplete(CommandSender sender, Command command, String label, String[] args) {
        if (args.length == 1) {
            return List.of("start", "stop", "start_hard_mode", "resume", "end", "nuke_area");
        }
        return List.of();
    }
}

File: src/main/java/base/minigames/disco_mayhem/DiscoMayhemCommands.kt
// src/main/java/me/stavgordeev/plugin/commands/MinigameCommand.java
package base.minigames.disco_mayhem

import base.commands.MinigameCommandsSkeleton
import base.minigames.maze_hunt.MazeHuntCommands
import org.bukkit.command.Command
import org.bukkit.command.CommandSender
import org.bukkit.entity.Player

class DiscoMayhemCommands(private val discoMayhem: DiscoMayhem) : MinigameCommandsSkeleton() {
    /**
     * All sub-commands for this minigame
     */
    enum class SubCommands {
        START,
        START_HARD_MODE,
        PAUSE,
        RESUME,
        END,
        NUKE_ARENA;

        companion object {
            /**
             * Converts a string to a SubCommand enum value. Case-insensitive.
             * @param str The string to convert
             * @return The SubCommand enum value, or null if the string does not match any enum value
             */
            fun fromString(str: String): SubCommands? {
                return entries.find { it.name.equals(str, ignoreCase = true) }
            }
        }
    }


    override fun handleCommand(sender: Player, command: Command, label: String, args: Array<String>): Boolean {
     when (SubCommands.fromString(args[0])) {
            SubCommands.START -> {
                if (discoMayhem.stopIfGameIsRunning()) return false
                discoMayhem.start(sender)
            }
            SubCommands.START_HARD_MODE -> {
                if (discoMayhem.stopIfGameIsRunning()) return false
                discoMayhem.startFastMode(sender)
            }
            SubCommands.PAUSE -> {
                if (discoMayhem.stopIfGameIsPaused()) return false
                discoMayhem.pauseGame()
            }
            SubCommands.RESUME -> {
                if (discoMayhem.stopIfGameIsNotPaused()) return false
                discoMayhem.resumeGame()
            }
            SubCommands.END -> {
                if (discoMayhem.stopIfGameIsNotRunning()) return false
                discoMayhem.endGame()
            }
            SubCommands.NUKE_ARENA -> discoMayhem.nukeArea(DiscoMayhemConst.GAME_START_LOCATION,DiscoMayhemConst.NUKE_AREA_RADIUS)
            else -> return error(sender, "Unknown command.")
        }

        return true
    }

    override fun onTabComplete(
        sender: CommandSender,
        command: Command,
        label: String,
        args: Array<String>
    ): List<String> {
        return when (args.size) {
            1 -> MazeHuntCommands.SubCommands.entries.map { it.name.lowercase()}
            else -> {listOf()}
        }
    }
}